File: src/Implementations/HandCuffs.cpp
#include "HandCuffs.h"
#include "GameState.h"

void HandCuffs::UseItem(GameState& gameState) 
{ 
	gameState.gameStateManager.SetStateOfHandCuffs(true);
	if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
	{
		gameState.log.push_back("Uzyles kajdanek!");
	}
	else
	{
		gameState.log.push_back("Komputer uzyl kajdanek!");
	}
}

File: C#/src/Implementations/HandCuffs.cs
public class HandCuffs : Item
{
    public HandCuffs() { }

    public void UseItem(GameState gameState)
    {
        gameState.gameStateManager.SetStateOfHandCuffs(true);
        if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
        {
            gameState.log.Add("Uzyles kajdanek!");
        }
        else
        {
            gameState.log.Add("Komputer uzyl kajdanek!");
        }
    }
}